lib/parse.js

```diff
@@ -33,7 +33,9 @@ module.exports = {
 
   asYaml: function (spec) {
     try {
-      let obj = yaml.safeLoad(spec);
+      let obj = yaml.safeLoad(spec, {
+        schema: yaml.JSON_SCHEMA
```

Comment:
Without this, the parser used to convert a date string to JS's `Data` object. Which was causing the validation to fail, since schema used to indicate the data should be of type `date` but it used to be `date-time` because of the type casting done by this parser.